src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -67,22 +67,37 @@
       final Collection<Unit> transports) {
 
     final List<Unit> canBeTransported = sortByTransportCostDescending(units);
-    final List<Unit> canTransport = sortByTransportCapacityAscendingThenMovesDescending(transports);
+    final List<Unit> canTransport = sortByTransportCapacityDescendingThenMovesDescending(transports);
 
     // Add max units to each transport
     final Map<Unit, Unit> mapping = new HashMap<>();
+    Optional<Unit> finalTransport = Optional.empty();
     for (final Unit transport : canTransport) {
-      int capacity = TransportTracker.getAvailableCapacity(transport);
-      for (final Iterator<Unit> it = canBeTransported.iterator(); it.hasNext();) {
-        final Unit unit = it.next();
-        final int cost = UnitAttachment.get((unit).getType()).getTransportCost();
-        if (capacity >= cost) {
-          capacity -= cost;
-          mapping.put(unit, transport);
-          it.remove();
+
+      // Check if remaining units can all be loaded into 1 transport
```

Comment:
One states "each" transport, then we are doing something that could only impact "1 transport", with an obvious loop break that would prevent us from doing something to "each" transport.

Both comments indicate serious smells, as they paraphrase what the code is doing, which should be self-evident.

Basically it's as if the second comment is restating the definition of the for loop... which is really not helpful. A "first check if we can load all units into 1 transport" would be better, but that still reads weird with a seemingly different game plan stated two lines earlier.